api/v1beta1/openstackvmset_types.go

```diff
@@ -42,8 +42,10 @@ type OpenStackVMSetSpec struct {
 	OSPNetwork Network `json:"ospNetwork"`
 	// Networks the name(s) of the OpenStackNetworks used to generate IPs
 	Networks []string `json:"networks"`
-	// Role the name of the Overcloud role this IPset is associated with. Used to generate hostnames.
+	// Role the name of the Overcloud role this RoleSpec is associated with. If it is a TripleO role, the name must match.
 	Role string `json:"role"`
+	// in case of external functionality, like 3rd party network controllers, set to false to ignore role in rendered overcloud templates.
+	IsTripleoRole bool `json:"isTripleoRole"`
```

Comment:
Instead of IsTripleoRole we might consider calling this ValidateRole or something more generic.

Reply:
we can find a better name for it. the main reason for this is to prevent 3rd party roles to be rendered into the tripleo environment files, like per deployed server port map or role count.